cset-workflow/flow.cylc

```diff
@@ -78,64 +111,46 @@ URL = https://metoffice.github.io/CSET-workflow
         [[[environment]]]
         CLEAN_WEB_DIR = {{CLEAN_WEB_DIR}}
 
-    [[finish_website]]
-    # Updates the workflow info in the web interface.
-    platform = localhost
-        [[[environment]]]
-
     [[fetch_fcst]]
     # Fetch data from disk or a file based archival system.
+    inherit = FETCH_DATA
         [[[environment]]]
         ROSE_APP_OPT_CONF_KEYS = {{FETCH_FCST_OPT_CONF}}
-        FILE_PATH = {{CSET_INPUT_FILE_PATH}}
+        CSET_INPUT_FILE_PATH = {{CSET_INPUT_FILE_PATH}}
+        {% if CSET_INCREMENTAL_DATA_FETCH %}
+        CSET_FILE_NAME_METADATA_PATTERN = {{CSET_FILE_NAME_METADATA_PATTERN}}
+        CSET_CYCLE_PERIOD = {{CSET_CYCLE_PERIOD}}
+        CSET_TIMES_PER_FILE = {{CSET_TIMES_PER_FILE}}
+        CSET_FILE_TIME_OFFSET = {{CSET_FILE_TIME_OFFSET}}
+        {% endif %}
 
-    [[recipe_start]]
-    platform = localhost
-    script = true
+    [[housekeeping_raw]]
+    # Housekeep unprocessed data files.
+    script = rose task-run -v --app-key=housekeeping
+        [[[environment]]]
+        HOUSEKEEPING_MODE = {{[HOUSEKEEPING_MODE, 1]|min}}
+
+    [[housekeeping_full]]
+    # Housekeep processed intermediate files too.
+    script = rose task-run -v --app-key=housekeeping
+        [[[environment]]]
+        HOUSEKEEPING_MODE = {{[HOUSEKEEPING_MODE, 2]|min}}
 
-    [[recipe_finish]]
+    [[finish_website]]
+    # Updates the workflow info in the web interface.
     platform = localhost
-    script = true
 
     [[send_email]]
     # Send email to notify that the workflow is complete.
     platform = localhost
         [[[environment]]]
         WEB_ADDR = {{WEB_ADDR}}
 
-    [[housekeeping]]
-        [[[environment]]]
-        HOUSEKEEPING_MODE = {{HOUSEKEEPING_MODE}}
-
 
 # Include files bring their own graph and runtime sections.
-{% if PLOT_SPATIAL_SURFACE_MODEL_FIELD %}
-{% include 'includes/plot_spatial_surface_model_field.cylc' %}
-{% endif %}
-{% if METPLUS_POINT_STAT %}
-{% include 'includes/point_stat.cylc' %}
-{% endif %}
-{% if DETERMINISTIC_PLOT_SURFACE_AIR_TEMP %}
-{% include 'includes/deterministic_plot_surface_air_temp.cylc' %}
-{% endif %}
-{% if DETERMINISTIC_PLOT_AGGREGATE_PRECIPITATION %}
-{% include 'includes/deterministic_plot_aggregate_precipitation.cylc' %}
-{% endif %}
-{% if DETERMINISTIC_PLOT_CAPE_RATIO %}
-{% include 'includes/deterministic_plot_cape_ratio.cylc' %}
-{% endif %}
-{% if DETERMINISTIC_PLOT_MODEL_LEVEL_AIR_TEMP %}
-{% include 'includes/deterministic_plot_model_level_air_temp.cylc' %}
-{% endif %}
-{% if ENSEMBLE_PLOT_SURFACE_AIR_TEMP %}
-{% include 'includes/ensemble_plot_surface_air_temp.cylc' %}
-{% endif %}
-{% if MEAN_HOURLY_ACCUMULATED_PRECIPITATION_ENSEMBLE_POSTAGE_STAMP_PLOT %}
-{% include 'includes/mean_hourly_accumulated_precip_ensemble_postage_stamp_plot.cylc' %}
-{% endif %}
-
+{% for include_file in glob("includes/*.cylc") %}
+{% include include_file %}
+{% endfor %}
 
 {# Site-specific details that add to (or override) the core suite definition #}
-{% if SITE %}
```

Comment:
If the site is not set then the workflow defaults to running on the scheduler host, as we found out after we ran a cylc server out of memory. So in #431 it was specifically made a required variable.

There is a localhost option for small scale testing, but it now has to be explicitly selected rather than being the default.